scanner/src/main/kotlin/Scanner.kt

```diff
@@ -141,15 +141,72 @@ class Scanner(
             }
         }
 
+        val filesByProvenance = (projectResults.files + packageResults.files)
+            .groupBy { it.provenance }
+            .values
+            .map { fileLists ->
+                fileLists.reduce { acc, next ->
+                    acc.copy(
+                        files = acc.files + next.files
+                    )
+                }
+            }
+            .toSet()
+
+        val deduplicatedScanResults = (projectResults.scanResults + packageResults.scanResults)
+            .groupBy { it.provenance to it.scanner }
+            .values
+            .map { scanResults ->
+                scanResults.reduce { acc, next ->
+                    acc.copy(
+                        summary = acc.summary.copy(
+                            startTime = minOf(acc.summary.startTime, next.summary.startTime),
+                            endTime = maxOf(acc.summary.endTime, next.summary.endTime),
```

Comment:
note, while this keeps correct `start` and `end`, it likely leads to wrong `durations`.